### DISCUSSION TITLE
JSON Format

### DISCUSSION THREAD

**Discussion by torstenschramm** [grafana/grafana-infinity-datasource#571] [Q&A] [answered]
Hello,

I have the following JSON Data (weather station):

[{"ts_id": "20078010","rows": "6","columns":"Timestamp,Value", "data": [["2023-03-30T00:00:00.000+01:00",4.94],["2023-03-31T00:00:00.000+01:00",15.53],["2023-04-01T00:00:00.000+01:00",27.07],["2023-04-02T00:00:00.000+01:00",0.84],["2023-04-03T00:00:00.000+01:00",0.00],["2023-04-04T00:00:00.000+01:00",0.00]]}]

With the default Parser and Format "As Is" i can visualize the data. But with the Backend Parser i have no success.

Have you any idea or can you help me ?

Thank you

**Comment by yesoreyeram** [comment, marked as answer]
backend parser doesn't understand the format correctly.. So you need some help with root selector/jsonata as shown below. Root selector used is `$map($[0].data,function($d){ return { 'timestamp': $d[0], 'value': $d[1] } })`

<img width="1792" alt="image" src="https://user-images.githubusercontent.com/153843/229784170-60ff8d83-504e-4d78-badb-c6249685b9ee.png">

**Reply by torstenschramm** [reply]
You are my hero. Thank you
